### PR TITLE
IU Java Utilities 7.0.2

### PR DESCRIPTION
**Pull request by mwfyffeiu**

### CODE REVIEW

**Review by james-anderson-iu — COMMENTED** [review]
Some comments from https://github.com/indiana-university/iu-java-util/pull/42 may apply here too and are not copied onto this PR.

File: base/src/main/java/edu/iu/IuBadRequestException.java [hunk 1]
```diff
@@ -29,46 +29,54 @@
...
-
-}
+package edu.iu;
+
+/**
+ * Thrown by an application or incoming request handler to represent an invalid
+ * incoming requests.
+ * 
+ * <p>
+ * <em>Should</em> be caught and handled as 400 BAD REUEST by an outbound web
```
Comment (mramawat) [inline base/src/main/java/edu/iu/IuBadRequestException.java:39]: fix type REQUEST

File: base/pom.xml [hunk 2]
```diff
@@ -7,12 +7,16 @@
...
 		<artifactId>iu-java-parent</artifactId>
-		<version>7.0.1</version>
+		<version>7.0.2-SNAPSHOT</version>
 	</parent>
 
 	<artifactId>iu-java-base</artifactId>
 	<name>IU Base Utilities Module</name>
 
+	<properties>
+		<argLine>-ea -Xmx1g --add-modules ALL-MODULE-PATH -XX:+EnableDynamicAgentLoading</argLine>
```
Comment (mramawat) [inline base/pom.xml:17]: @mwfyffeiu  You may have explained this before but forget why we need to EnableDynamicAgentLoading. Java future release will disallow this feature as a default https://openjdk.org/jeps/451.

File: base/src/main/java/edu/iu/IuOutOfServiceException.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+ */
+package edu.iu;
+
+/**
+ * Thrown by an application or incoming request handler to represent a session
+ * activation failure due to a <em>planned</em> downstream service or
+ * application outage.
+ * 
+ * <p>
+ * <em>Should</em> be caught and handled as 503 SERVICE UNVAILABLE by an
```
Comment (mramawat) [inline base/src/main/java/edu/iu/IuOutOfServiceException.java:40]: fix type UNVAILABLE

File: base/src/main/java/edu/iu/IuWebUtils.java [hunk 4]
```diff
@@ -0,0 +1,789 @@
...
+	 * @param c character
+	 * @return true if c matches obs-text ABNF rule; else false
+	 * @see <a href=
+	 *      "https://www.rfc-editor.org/rfc/rfc9110.html#name-collected-abnf">RFC-9110
+	 *      HTTP Semantics Collected ABNF</a>
+	 */
+	static boolean tchar(char c) {
+		return alpha(c) //
+				|| digit(c) //
+				|| "!#$%&'*+-.^_`!~".indexOf(c) != -1;
```
Comment (mramawat) [inline base/src/main/java/edu/iu/IuWebUtils.java:163]: seems like second ! mark should be |

**Pull request merged**